### ISSUE TITLE
EntityContainer doesn't support nested parametrized types

### ISSUE THREAD

**Issue by MeFisto94** [Simsilica/SiO2#1]
Using
```
public class RigidBodyContainer extends EntityContainer<Tuple<Spatial, RigidBodyControl>>
```
leads to the following Exception:
```
java.lang.ClassCastException: sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl cannot be cast to java.lang.Class
        at com.simsilica.es.EntityContainer.<init>(EntityContainer.java:96)
        at com.simsilica.es.EntityContainer.<init>(EntityContainer.java:83)
        at de.darkigames.TheSlaughter.EntitySystem.Util.RigidBodyContainer.<init>(RigidBodyContainer.java:26)
```

**Comment by Ali-RS** [comment]
Hi,
Yet any plan to add support for this ?
Thought to remind it :)

**Comment by assofohdz** [comment]
Just linking to a hub post here https://hub.jmonkeyengine.org/t/sio2-entitycontainer-and-parameterizedtype-exception/39182

**Comment by MeFisto94** [comment]
Looks like the workaround to this problem was only posted on the hub.
A simple dummy class `class A extends Tuple<Spatial, RigidBodyControl>` should do the trick, iirc.

**Comment by Ali-RS** [comment]
Should be fixed now in this commit : 75decab8fe25fe49cf42ff4a264f0ce6b7c57ab1
I think we can close this now @MeFisto94
